src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -346,6 +378,37 @@ public ProgrammingExerciseResource(ProgrammingExerciseRepository programmingExer
         return ResponseUtil.wrapOrNotFound(programmingExercise);
     }
 
+    /**
+     * GET /programming-exercises-with-participations/:id : get the "id" programmingExercise.
+     *
+     * @param id the id of the programmingExercise to retrieve
+     * @return the ResponseEntity with status 200 (OK) and with body the programmingExercise, or with status 404 (Not Found)
+     */
+    @GetMapping("/programming-exercises-with-participations/{id}")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ProgrammingExercise> getProgrammingExerciseWithAllParticipations(@PathVariable Long id) {
+        log.debug("REST request to get ProgrammingExercise : {}", id);
+        Optional<ProgrammingExercise> programmingExerciseOpt = programmingExerciseRepository.findWithAllParticipationsById(id);
+        if (programmingExerciseOpt.isPresent()) {
+            ProgrammingExercise programmingExercise = programmingExerciseOpt.get();
+
+            // Make sure to not return the template/solution participation in the participations array
+            Set<Participation> participations = new HashSet<>();
+            programmingExercise.getParticipations().stream().filter(p -> p.getStudent() != null).findFirst().ifPresent(participations::add);
```

Comment:
I noticed that this endpoint is only used by the code editor for the instructor. For the instructor only one assignment participation exists, so this implementation is probably fine. However, it might lead to problems when there already are real student participations. But I am not sure, if we have to cover this case.

Reply:
True, I'll fix it.